internal/imagedestination/impl/helpers.go

```diff
@@ -0,0 +1,19 @@
+package impl
+
+import (
+	"github.com/containers/image/v5/internal/private"
+	compression "github.com/containers/image/v5/pkg/compression/types"
+)
+
+// BlobMatchesRequiredCompression validates if compression is required by the caller while selecting a blob, if it is required
+// then function performs a match against the compression requested by the caller and compression of existing blob.
+func BlobMatchesRequiredCompression(options private.TryReusingBlobOptions, originalCompression *compression.Algorithm) bool {
```

Comment:
```suggestion
func BlobMatchesRequiredCompression(options private.TryReusingBlobOptions, blobCompression *compression.Algorithm) bool {
```
or `candidateCompression`… the Docker transports calls it with replacement candidates, not always with the original one we are trying to replace.